sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerReadSchemaTransformProvider.java

```diff
@@ -159,13 +159,13 @@ public void validate() {
         checkNotNull(this.getColumns(), invalidConfigMessage + "Columns must be specified for table read.");
       }
       else {
-        checkNotNull(this.getQuery(), invalidConfigMessage + "Query must be specified for query read.");
+        checkNotNull(Strings.isNullOrEmpty(this.getQuery()), invalidConfigMessage + "Query must be specified for query read.");
```

Comment:
This will never fail since `Strings.isNullOrEmpty(this.getQuery())` is a boolean (never null)
```suggestion
        checkNotNull(Strings.isNullOrEmpty(this.getQuery()), invalidConfigMessage + "Query must be specified for query read.");
```
```suggestion
        checkArgument(
          Strings.isNullOrEmpty(this.getQuery()),
          invalidConfigMessage + "Query must be specified for query read."
        );
```